routing/bicycle_directions.cpp

```diff
@@ -170,7 +185,13 @@ void BicycleDirectionsEngine::Generate(IRoadGraph const & graph, vector<Junction
   Route::TTimes turnAnnotationTimes;
   Route::TStreets streetNames;
 
-  MakeTurnAnnotation(resultGraph, delegate, routeGeometry, turns, turnAnnotationTimes, streetNames);
+  MakeTurnAnnotation(resultGraph, delegate, routeGeometry, turns, turnAnnotationTimes,
+                     streetNames, routeSegs);
+  // @TODO(bykoianko) The invariant below it's an issue but now it's so and it should be checked.
+  // The problem is every edge is added as a pair of point to route geometry.
+  // So all the points except for beginning and ending are duplicated. It should
+  // be fixed in the future.
+  ASSERT_EQUAL(routeGeometry.size(), 2 * routeSegs.size(), ());
```

Comment:
Are you sure? Number of points should be equal to number of segments + 1, isn't it?